conn2res/plotting.py

```diff
@@ -574,78 +600,40 @@ def plot_performance(
 
     # plot
     sns.lineplot(
-        data=df, x=x, y=y, hue=hue, palette=palette, markers=True,
-        legend=True, ax=ax, **kwargs)
-
-    # set axis labels
-    ax.set_xlabel('alpha', fontsize=11)
-    y_label = ' '.join(y.split('_'))
-    ax.set_ylabel(y_label, fontsize=11)
+        data=df, x=x, y=y, hue=hue, palette=palette, dashes=False,
+        legend=False, markers=True, ax=ax, **kwargs)
 
     # set legend
-    plt.legend(loc='upper right', fontsize=8)
+    try:
+        lg_defaults = {'labels': kwargs['hue_order']}
+    except:
+        lg_defaults = {'labels': list(pd.unique(df[hue]))}
+    lg_defaults.update(**lg_params)
+    ax.legend(handles=ax.lines, **lg_defaults)
+
+    # set axis properties
+    axes_defaults = {'xlabel': x, 'ylabel': ' '.join(y.split('_'))}
+    if title is not None:
+        axes_defaults.update(title=title)
+    axes_defaults.update(**ax_params)
+    ax.set(**axes_defaults)
 
     # set title
-    if title is not None:
-        plt.title(title, fontsize=12)
+    # if title is not None:
+    #     fig.suptitle(title)
 
     sns.despine(offset=10, trim=True,
                 top=True, bottom=False,
                 right=True, left=False)
 
     if show:
-        plt.show()
+        plt.show(block=True)
 
     if savefig:
-        if fname is None:
-            fname = 'performance_curve'
-
-        fig.savefig(fname=os.path.join(fig_dir, f'{fname}.png'),
+        fig.savefig(fname=fname + '.' + mpl.rcParams['savefig.format'],
                     transparent=True, bbox_inches='tight', dpi=300)
 
     plt.close()
 
-
-def plot_phase_space(x, y, sample=None, xlim=None, ylim=None, subplot=None, cmap=None,
-    num=1, figsize=(13, 5), title=None, fname='phase_space', savefig=False, block=False
-):
```

Comment:
I deleted this chunk of code as I believe it is only useful for my specific plots and it uses matplotlib plots instead of seaborn plots so it is not consistent with the rest of the plotting functions.

For similar reasons, even though I added functions to my local repository to be able to reproduce how I normally plot figures but I believe they are not useful for the toolbox so I kept them only locally on a different branch.

Reply:
I actually wanted to keep this function and make it using the same format of previous figures, because I think it is cool 😂

Reply:
Ok-ok, let's keep it, I have already revised this function and will update the pull request with that then.